pandas/core/strings.py

```diff
@@ -2400,13 +2411,17 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
         'side': 'beginning',
         'method': 'split'})
     def split(self, pat=None, n=-1, expand=False):
+        if self._inferred_type in ['bytes']:
```

Comment:
make this a method instead and call it where appropriate (or can do a decorator)

```
def assert_bytes_allowed(self):
   ....
```